### DISCUSSION TITLE
Using Yubikey 2FA with SEB

### DISCUSSION THREAD

**Discussion by AmirAttoun** [SafeExamBrowser/seb-win-refactoring#648] [General]
Hi 

Some of our students are using Yubikey hardwaretokens for 2FA (Microsoft 365).
Since we run exams through Moddle, which in our case requires the students to logon through M365 and using 2FA, we run into trouble. The students can't use the Hardwaretokens as 2FA inside SEB. We get to the stage, where Microsoft tries to open an additional window for authentication via security key. The windows never opens. We tried whitelisting and allowing pop ups.

Has anyone successfully used a Yubikey (or similar) with SEB as a method of 2FA?

Thanks
Amir

**Comment by dbuechel** [comment]
The client and/or browser logs of the respective sessions may provide useful information, see `%LocalAppData%\SafeExamBrowser\Logs`. You could also (for debugging purposes) allow the live application log and see what happens when trying to authenticate, see https://safeexambrowser.org/windows/win_usermanual_en.html#SecurityPane > "Show log button in taskbar".

**Comment by kmh42** [comment]
@AmirAttoun did you solve the problem?
We ran into the same problem with Microsoft 2FA. The 2FA windows opens up in the background with no why to see it in SEB.

**Comment by AntonT76** [comment]
Hi @AmirAttoun, @kmh42, @dbuechel 

we are also using multi factor authentication and normaly we recommend to use Microsoft Authenticator App. Some of our students are using a hardware key (eg Yubikey) - we also have some problems, but after some tests it will work on Windows 11 and SEB 3.9. You have to set a few tings in SEB config:

Security:
- Kiosk Mode - Disable Explorer Shell
- SEB Service Policy - allow to use SEB only with service

Applications:
- allow following EXE-File: CredentialUIbroker.exe (this might be the pop up window which shows you the steps for using Yubikey, is located in windows/system32)

I'm not sure what effect this two settings in the “Security” tab have. Perhaps someone can provide a detailed explanation?

thanks, br Anton

**Reply by dbuechel** [reply]
These are explained in the documentation, please refer to https://safeexambrowser.org/windows/win_usermanual_en.html#SecurityPane.
